scripts/etc/init.d/rcS

```diff
@@ -533,66 +533,66 @@ fi
 echo "Starting install process..."
 cdebootstrap-static --arch=armhf $cdebootstrap_cmdline $release /rootfs $mirror --keyring=/usr/share/keyrings/raspbian-archive-keyring.gpg || fail
 
-# allow root login
 echo "Configuring installed system:"
-if [ ! "$disable_root" = "1" ]; then
-    echo -n "  Setting root password... "
-    echo -n root:$rootpw | chroot /rootfs /usr/sbin/chpasswd || fail
-    echo "OK"
-fi
 
-# add SSH key for root (if provided)
-if [ ! -z "$root_ssh_pubkey" ]; then
-    echo -n "  Setting root SSH key... "
-    mkdir -p /rootfs/root/.ssh && \
-        chmod 700 /rootfs/root/.ssh && \
-        echo "$root_ssh_pubkey" > /rootfs/root/.ssh/authorized_keys
-    if [ $? -eq 0 ]; then
+# configure root login
+if [ ! "$disable_root" = "1" ]; then
+    # add SSH key for root (if provided)
+    if [ ! -z "$root_ssh_pubkey" ]; then
+        echo -n "  Setting root SSH key... "
+        mkdir -p /rootfs/root/.ssh && \
+            echo "$root_ssh_pubkey" > /rootfs/root/.ssh/authorized_keys || fail
+        echo "OK"
+        echo -n "  Setting permissions on root SSH directory... "
+        chmod -R go-rwx /rootfs/root/.ssh || fail
```

Comment:
The `chmod 700` on the `~/.ssh` directory is better (and iirc that's what ssh checks for).
On various systems, I have files in the `~/.ssh/` directory which are world-readable and while your code doesn't seem to practically hinder that _now_, it does give the wrong impression regarding permissions of files in the `~/.ssh/` directory and could result in issues if code was added before your `chmod` statement.
I think the cleanest way for this code would be:
```sh
mkdir -p -m 700 /rootfs/root/.ssh || fail
echo "$root_ssh_pubkey" > /rootfs/root/.ssh/authorized_keys || fail
```
(and the same with the user creation)